src/stt/src/flt/flute.cpp

```diff
@@ -114,12 +114,12 @@ static void readLUTfiles(LUT_TYPE LUT, NUMSOLN_TYPE numsoln)
         fscanf(fpwv, "%d", &kk);
         fgetc(fpwv);  // '/n'
         numsoln[d][k] = numsoln[d][kk];
-        LUT[d][k] = LUT[d][kk];
+        (*LUT)[d][k] = (*LUT)[d][kk];
       } else {
         fgetc(fpwv);  // '\n'
         numsoln[d][k] = ns;
-        p = (struct csoln*) malloc(ns * sizeof(struct csoln));
-        LUT[d][k] = p;
+        p = new struct csoln[ns];
+        (*LUT)[d][k] = std::shared_ptr<struct csoln[]>(p);
```

Comment:
Better to use std::make_shared

Reply:
If I remember correctly I tried exactly that, but had issues making `std::make_shared` behave as I wanted, which is why I decided to use that solution, since in any case I need access to the raw pointer for the rest of the code to work

Reply:
I pushed a change that I think is an improvement